Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -141,76 +147,80 @@ std::string DecodeString(pb_istream_t* stream) {
   return result;
 }
 
-}  // namespace
-
-using firebase::firestore::model::FieldValue;
-
-void Serializer::EncodeFieldValue(const FieldValue& field_value,
-                                  std::vector<uint8_t>* out_bytes) {
-  // TODO(rsgowman): how large should the output buffer be? Do some
-  // investigation to see if we can get nanopb to tell us how much space it's
-  // going to need.
-  uint8_t buf[1024];
-  pb_ostream_t stream = pb_ostream_from_buffer(buf, sizeof(buf));
+void EncodeObject(
+    pb_ostream_t* stream,
+    const std::map<const std::string, const FieldValue>& object_value);
+std::map<const std::string, const FieldValue> DecodeObject(
+    pb_istream_t* stream);
 
+// Named '..Impl' so as to not conflict with Serializer::EncodeFieldValue.
```

Comment:
This isn't a useful comment. It focuses on why the names are different without drawing a distinction between the differences in _behavior_ which is the actually interesting part.

Consider two audiences:
  * The first-time reader wading in here for the first time: help them tame the complexity in this file. Help understand the differences between these two. Why would I choose one or the other?
  * The over-zealous new project member that wants to refactor everything: why are these separate? Do they need to be? What breaks if you use the wrong one?